src/python/grpcio/grpc/_common.py

```diff
@@ -97,3 +98,27 @@ def serialize(message, serializer):
 def deserialize(serialized_message, deserializer):
   return _transform(serialized_message, deserializer,
                     'Exception deserializing message!')
+
+
+class CleanupThread(threading.Thread):
+  """A wrapper around thread for custom join behavior.
+  
+  On Interpreter exit, Python will attempt to join outstanding threads
+  prior to garbage collection.  We may need to do additional cleanup, and
+  we acomplish this by wrapping the join() method.
+  """
+  
+  def __init__(self, group=None, target=None, join_target=None, name=None,
+               args=(), join_args=(), kwargs={}, join_kwargs={}):
+    """join_target() will be called with a timeout,
```

Comment:
Actually those examples in the style guide don't go far enough to illustrate what I'm saying. :-P